app/src/main/java/org/p2p/wallet/infrastructure/network/interceptor/RpcInterceptor.kt

```diff
@@ -30,7 +45,54 @@ class ServerErrorInterceptor(
         }
     }
 
-    private fun handleResponse(response: Response): Response {
+    private val requestBuffer = Buffer()
+    private fun createRpcRequest(chain: Interceptor.Chain): Request {
+        val request = chain.request()
+
+        val json = getRequestJson(request)
+
+        // Refactor this
+        val url = request.url
+
+        val httpUrl = if (json.getString("method") == "getConfirmedTransaction") {
+            url.newBuilder().host(getBaseUrl(Environment.RPC_POOL)).build()
+        } else {
+            url.newBuilder().host(getBaseUrl(currentEnvironment)).build()
+        }
+
+        return request.newBuilder().url(httpUrl).build()
+    }
+
+    private fun getRequestJson(request: Request): JSONObject {
+        request.body?.writeTo(requestBuffer)
+
+        val requestBodyString = requestBuffer.readUtf8()
+        return if (requestBodyString.startsWith("[")) {
+            val jsonArray = JSONArray(requestBodyString)
+            jsonArray.getJSONObject(0)
+        } else {
+            JSONObject(requestBodyString)
+        }
+    }
+
+    private fun getBaseUrl(environment: Environment): String {
+        val endpoint = when (environment) {
+            Environment.MAINNET -> environment.endpoint
+            Environment.SOLANA -> environment.endpoint
+            Environment.DEVNET -> environment.endpoint
```

Comment:
from our code style, if you have brackets for one branch - for consistency you need to add brackets for other branches too. 
Or extract Environment.RPC_POOL to a method and remove brackets